src/ui/control/navigation_control.js

```diff
@@ -78,6 +79,7 @@ class NavigationControl {
         DOM.remove(this._container);
         if (this.options.showCompass) {
             this._map.off('rotate', this._rotateCompassArrow);
+            DOM.removeEventListener(this._compass, 'mousedown', this._handler.onMouseDown.bind(this._handler));
```

Comment:
This won't actually remove anything because `bind` returns a new function, and the newly crested listener isn't among the added ones for the event. You should have `onMouseDown` already bound in the handler instead.